test/header/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@
 # shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
 # -----------------------------------------------------------------------------------------------------
 
-cmake_minimum_required (VERSION 3.2)
+cmake_minimum_required (VERSION 3.6)
```

Comment:
Have you checked that this version is compliant with all platforms that also support at least gcc-7?

Reply:
I upped the version, which we already implicitly required. Because we already use the following feature which is available since 3.6

```cmake
list (FILTER header_files EXCLUDE REGEX "${exclude_regex}")
```

Since all builds go through since forever, this should have no impact

Reply:
did you update the documentation?

Reply:
No, I didn't update the documentation. Or, do we want to set the minimal required version of cmake to 3.6? I'm happy to do that, but that would be out the scope of the PR.

I added a card for the dev meeting to discuss a general increase of cmake version.

Reply:
See core-developers chat.

Reply:
I moved the fix of the cmake versions to PR #1094 , so we can discuss this over there